### ISSUE TITLE
UX: Fluent navigation. Building on top of "filter files in real-time", and popup placement.

### ISSUE THREAD

**Issue by adriangalilea** [sxyazi/yazi#1390]
### Please describe the problem you're trying to solve

Currently my navigation is absurdly slow in yazi compared to things like broot.

### Current navigation UX:

Press <kbd>/</kbd>

Find popup appears.
<img width="578" alt="image" src="https://github.com/user-attachments/assets/448b0116-9e37-4125-aef1-4e18ec9addaf">

You enter your keyword, sometimes it doesn't even match.

I have to:

<kbd>esc</kbd>

<kbd>esc</kbd> again, after I remember it needs to be pressed twice.

<kbd>?</kbd> because I forgot how to "find next"

<kbd>n</kbd> until I find it.

<kbd>enter</kbd>

Repeat enough times until you arrive to your destination.

**TL;DR**: unintuitive, defies my expectations several times, too many keypresses for just moving 1 level deeper.

### Would you be willing to contribute this feature?

- [ ] Yes, I'll give it a shot

### Describe the solution you'd like

Building on top of https://github.com/sxyazi/yazi/pull/454 I think the UX could be as follows:

<kbd>/</kbd> <-- redefining filter as default in this example.

filter files

<img width="584" alt="image" src="https://github.com/user-attachments/assets/8072dd1d-93ff-44e7-8b95-3fb0919fd40d">

Clear visual of how many matches, much better than default <kbd>/</kbd> (find)

### Problem:

I still need to <kbd>esc</kbd>x2 and select the right one(arrow keypresses), then press <kbd>enter</kbd>

This is completely unnecessary.

Imagine we are at this point where we filtered out all the files/directories enough to our liking,  like in pic attached.

### Proposing fluent navigation UX inside `filter` mode:

<kbd>tab</kbd> inside the popup to change focus between remaining elements.

<kbd>/</kbd> to confirm selection without exiting filter mode.

Note, `/` is a safe key to use in this mode because it can't ever be used inside a file/directory name, so we can use this as keybind it for "move into focused element" that is, the exact same behaviour as exiting and pressing enter on it.

### One more thing:

When you enter a directory like this, preserve the `filter` mode, just clean it from the previous text, so you can start filtering again, <kbd>tab</kbd> and <kbd>shift</kbd>-<kbd>tab</kbd> to change focus and <kbd>/</kbd>  to confirm selection until you arrive to your destination.

**Bonus:** Remove the popup for filtering files and place it on the "status bar" or in the "header"

### Additional context

I think this proposal would benefit from visuals, but unfortunately I can't do it now, may do it in the future if my idea is not clear enough.

**Comment by sxyazi** [comment]
> When you enter a directory like this, preserve the `filter` mode, just clean it from the previous text, so you can start filtering again, <kbd>tab</kbd> and <kbd>shift</kbd>-<kbd>tab</kbd> to change focus and <kbd>/</kbd>  to confirm selection until you arrive to your destination.

This really sounds like the kind of work the smart-filter plugin does - it will automatically enter the only directory, and I've also added other enhancements, like supporting continuous filtering and automatically opening the hovered file on submitting:


https://github.com/sxyazi/yazi/assets/17523360/0007e5e5-ed9e-4104-ad13-0678b73af1f1


Here's the code (less than 50 lines): https://github.com/yazi-rs/plugins/tree/main/smart-filter.yazi, feel free to use it or build your own workflow based on it.

You can find more awesome plugins at https://yazi-rs.github.io/docs/resources

Closing as there's already a plugin with very similar functionality, and this is highly specific to a particular workflow, so it's better suited as a plugin implementation.

**Comment by adriangalilea** [comment]
Awesome, that's pretty much it, thanks!

> Bonus: Remove the popup for filtering files and place it on the "status bar" or in the "header"

Any option for this?

Can we change the location of the popup? Place it on the status bar or  header?

**Comment by sxyazi** [comment]
> Can we change the location of the popup? 

See https://yazi-rs.github.io/docs/configuration/yazi#input

**Comment by adriangalilea** [comment]
@sxyazi I'm trying to adapt your init.lua to match my desired UX, but I have no idea what events are available to me or where are they defined? How can I listen for:

<kbd>/</kbd> to accept focus, like you got enter
<kbd>tab</kbd> to change focus to next
<kbd>shift-tab</kbd> to change focus to prev
<kbd>backspace</kbd> (if no text in input) to go to prev folder

inside the input window? I wasn't able to make any of those work.

I'm not familiar with lua, nor rust, and even with the help of claude and code-hunting for those event codes I could not find what I'm missing, would appreciate your input, will keep trying in the meantime, thanks.
